quota/etcd/README.md

```diff
@@ -0,0 +1,246 @@
+# Etcd quotas
+
+Package etcd (and its subpackages) contain an etcd-based
+[quota.Manager](https://github.com/google/trillian/blob/3cf59cdfd0/quota/quota.go#L101)
+implementation, with a corresponding REST-based configuration service.
+
+## Usage
+
+First, ensure both logserver and logsigner are started with the `--etcd_servers`
+and `--quota_system=etcd` flags, in addition to other flags. Logserver must
+also be started with a non-empty `--http_endpoint` flag, so the REST quota API
+is bound.
+
+For example:
+
+```bash
+trillian_log_server \
+  --etcd_servers=... \
+  --http_endpoint=localhost:8091 \
+  --quota_system=etcd
+
+trillian_log_signer --etcd_servers=... --quota_system=etcd
+```
+
+If correctly started, the servers will be using etcd quotas. The default
+configuration is empty, which means no quotas are enforced.
+
+The REST quota API may be used to create and update configurations.
+
+For example, the command below creates a sequencing-based, global/write quota.
+Assuming an expected sequencing performance of 50 QPS, the max_tokens specified
+implies a backlog of 4h.
+
+```bash
+curl \
+  -d '@-' \
+  -s \
+  -H 'Content-Type: application/json' \
+  -X POST \
+  "localhost:8091/v1beta1/quotas/global/write/config" <<EOF
+{
+  "name": "quotas/global/write/config",
+  "config": {
+    "state": "ENABLED",
+    "max_tokens": 288000,
+    "sequencing_based": {
+    }
+  }
+}
+EOF
+```
+
+To list all configured quotas, do:
+
+```bash
+curl localhost:8091/v1beta1/quotas?view=FULL
+```
+
+Quotas may be retrieved individually or via a series of filters, updated and
+deleted through the REST API as well. See
+[quotapb.proto](https://github.com/google/trillian/blob/master/quota/etcd/quotapb/quotapb.proto)
+for an in-depth description of entities and available methods.
+
+### Maintenance and token exhaustion
+
+During regular system operation, no quota-related maintenance should be
+required, as the system should generate at least as many tokens as it spends.
+
+If a token outage happens there are a few built-in mechanisms that allow manual
+intervention. The question of whether intervention is needed, though, is an
+important one and should be answered before any attempts to bypass the system.
+For example:
+
+* Is logsigner working properly and able to keep with the current demand?
+* Is there a spike in requests that may justify the current token outage?
+
+In genuine outages, it may be beneficial to let the quota system deny requests
+until regular operation is resumed.
+
+That said, the sections below describe a few alternatives that may used to deal
+with token outages. All examples use **global/read** as the quota in question;
```

Comment:
outage -> depletion?

Reply:
Ack (addressed as part of a previous comment).